Program february (#52)
rust-aarhus/meetups: meetups/2024/2024-02-22-talks/program.md

# Rust Aarhus meetup at Partisia

Date: 2024-02-22

Rust Aarhus invites you to this free meetup for all interested in the programming language [Rust].

The program for the evening is:

- 18:05 - 18:10: Welcome (5 min)
- 18:15 - 19:00: "Building serverless apps, using WASI, WebAssembly components using Rust" by Mikkel Mørk Hegnhøj (45 min incl Q&A)
- 19:00 - 19:10: A word from our sponsor (5-10 min)

> - 19:10 - 19:55: Light dinner sponsored by Partisia (45 min)

Reviewer: Partisia is still figuring out who does their talk, so I think we should just add this, and we can update the agenda later with the actual speaker, if we want (for me it is also fine to just let it be as the company name).
I took 10 minutes from the lunch time, we still have 35 mins, which should be plenty time to eat a sandwich.
```suggestion
- 19:10 - 19:20: Using Rust at Partisia by Partisia (10 min)
- 19:20 - 19:55: Light dinner sponsored by Partisia (45 min)
```

- 19:55 - 20:40: "Zero-copy deserialization in Rust" by Gustav Wengel (45 min incl Q&A)
- 20:45 - 20:55: Book Lottery (10 min)
- 20:55 - 21:00: Closing remarks (5 min)

> - 21:00: Let's continue the discussion at the nearby bar [INSERT BAR NAME].

Reviewer: Lets omit this part about the bar for now, so we can get this published, then we can always update it later to add the bar.
```suggestion
```

## Zero-copy deserialization in Rust
By [Gustav Wengel][gw]

When using Rust in many contexts, such as webservers or just parsing structured data from files, we need to deserialize text or bytes into Rust types. The de-facto library for this in the Rust ecosystem is [serde] - and we're going to explore how serde allows us to deserialize objects without actually allocating memory, by borrowing from the original source. We'll also dive into the [rkyv] crate as a zero-copy deserialization library if you really need every last bit of performance.

Gustav has been working with Rust for the past two years, building serverless RESTful API's at the startup Climatiq, that allows the users to embed CO2 and climate calculations into existing software. Before that he has worked in different startups with different tech stacks, and has been a core maintainer of popular open-source projects like [ts-jest].

## Location

> This meetup will be at [INSERT ADDRESS HERE]. Please feel free to use the event chat on meetup to organize transportation if needed.

Reviewer: I am pretty sure the address is, but I am awaiting confirmation from Partisia:
INCUBA Katrinebjerg, Lille Auditorium
Åbogade 15
8200 Aarhus

Author: ok, I will wait for the final confirmation

Reviewer: Confirmed, The address is correct.

> This meetup will be at [INSERT ADDRESS HERE]. Please feel free to use the event chat on meetup to organize transportation if needed.

Author: ```suggestion
This meetup will be at INCUBA Katrinebjerg, Lille Auditorium, Åbogade 15, 8200 Aarhus. Please feel free to use the event chat on meetup to organize transportation if needed.
```
